site/src/docs/typechat-in-5-minutes.md

```diff
@@ -0,0 +1,147 @@
+---
+layout: doc-page
+title: TypeChat in 5 Minutes
+---
+
+Currently TypeChat's a very small library, so let's start breaking down a few functions to better understand how to use it.
+
+## Providing a Model
+
+TypeChat is mostly model agnostic.
+As long as you can construct an object with the following properties:
+
+```ts
+export interface TypeChatLanguageModel {
+    /**
+     * Optional property that specifies the maximum number of retry attempts (the default is 3).
+     */
+    retryMaxAttempts?: number;
+    /**
+     * Optional property that specifies the delay before retrying in milliseconds (the default is 1000ms).
+     */
+    retryPauseMs?: number;
+    /**
+     * Obtains a completion from the language model for the given prompt.
+     * @param prompt The prompt string.
+     */
+    complete(prompt: string): Promise<Result<string>>;
+}
+```
+
+then you should be able to try TypeChat out with such a model.
+The key thing here is that only the `complete` property is required, and is a function taking only a `string` and eventually returning a `string` if all goes well.
+
+For convenience, TypeChat provides two functions out of the box to connect to the OpenAI API and Azure's OpenAI Services.
+You can call these directly.
+
+```ts
+export function createOpenAILanguageModel(apiKey: string, model: string, endPoint? string): TypeChatLanguageModel;
+
+export function createAzureOpenAILanguageModel(apiKey: string, endPoint: string): TypeChatLanguageModel;
+```
+
+For even more convenience, TypeChat also provides a function to infer whether you're using OpenAI or Azure OpenAI.
+
+```ts
+export function createLanguageModel(env: Record<string, string | undefined>): TypeChatLanguageModel
+```
+
+You can populate your environment variables, and based on whether `OPENAI_API_KEY` or `AZURE_OPENAI_API_KEY` is set, you'll get a model of the appropriate type.
+
+```ts
+import dotenv from "dotenv";
+dotenv.config(/*...*/);
+import * as typechat from "typechat";
+const model = typechat.createLanguageModel(process.env);
+```
+
+Regardless, of how you decide to construct your model, we recommend keeping your secret tokens/API keys in a `.env` file, and specifying `.env` in a `.gitignore`.

```

Comment:
Could we provide an example of an alternative model other than OpenAI?

Reply:
I was kind of thinking of having a separate page for that with specific examples for each model. We could possibly use a dummy example here?

Reply:
> I was kind of thinking of having a separate page for that with specific examples for each model. We could possibly use a dummy example here? What do you think?

I think having an separeted page for this instructions will be amazing!